e2e/tests/functional/plugins/notebook/notebookWithCouchDB.e2e.spec.js

```diff
@@ -0,0 +1,232 @@
+        let requests = [];
+        page.on('request', (request) => requests.push(request));
+
+        let [notebookUrlRequest, allDocsRequest] = await Promise.all([
+            // Waits for the next request with the specified url
+            page.waitForRequest(`**/openmct/${testNotebook.uuid}`),
+            page.waitForRequest('**/openmct/_all_docs?include_docs=true'),
+            // Triggers the request
+            page.click('text=Page Add >> button'),
+            // Ensures that there are no other network requests
+            page.waitForLoadState('networkidle')
+        ]);
+        // Assert that only two requests are made
+        // Network Requests are:
+        // 1) The actual POST to create the page
+        // 2) The shared worker event from 👆 request
+        expect(requests.length).toBe(2);
+
+        // Assert on request object
+        expect(notebookUrlRequest.postDataJSON().metadata.name).toBe('TestNotebook');
+        expect(notebookUrlRequest.postDataJSON().model.persisted).toBeGreaterThanOrEqual(notebookUrlRequest.postDataJSON().model.modified);
+        expect(allDocsRequest.postDataJSON().keys).toContain(testNotebook.uuid);
+
+        // Add an entry
+        // Network Requests are:
+        // 1) The actual POST to create the entry
+        // 2) The shared worker event from 👆 POST request
+        requests = [];
+        await page.locator('text=To start a new entry, click here or drag and drop any object').click();
+        await page.locator('[aria-label="Notebook Entry Input"]').click();
+        await page.locator('[aria-label="Notebook Entry Input"]').fill(`First Entry`);
+        await page.waitForLoadState('networkidle');
+        expect(requests.length).toBeLessThanOrEqual(2);
+
+        // Add some tags
+        // Network Requests are for each tag creation are:
+        // 1) Getting the original path of the parent object
+        // 2) Getting the original path of the grandparent object (recursive call)
+        // 3) Creating the annotation/tag object
+        // 4) The shared worker event from 👆 POST request
+        // 5) Mutate notebook domain object's annotationModified property
+        // 6) The shared worker event from 👆 POST request
+        // 7) Notebooks fetching new annotations due to annotationModified changed
+        // 8) The update of the notebook domain's object's modified property
+        // 9) The shared worker event from 👆 POST request
+        // 10) Entry is timestamped
+        // 11) The shared worker event from 👆 POST request
+
+        requests = [];
+        await page.hover(`button:has-text("Add Tag")`);
+        await page.locator(`button:has-text("Add Tag")`).click();
+        await page.locator('[placeholder="Type to select tag"]').click();
+        await page.locator('[aria-label="Autocomplete Options"] >> text=Driving').click();
+        await page.waitForSelector('.c-tag__label:has-text("Driving")');
```

Comment:
would you mind looking at the a11y pattern for tags? I feel like this is somehow incorrect

Reply:
The tags take awhile to appear here due to CouchDB latency. This doesn't happen with localStorage.